tests/unit/amo/sagas/testUserAbuseReports.js

```diff
@@ -0,0 +1,153 @@
+    const message = 'I would prefer the user be XUL';
+    const response = createFakeUserAbuseReport({ message, userId });
+
+    mockApi
+      .expects('reportUser')
+      .once()
+      .returns(Promise.resolve(response));
+
+    _sendUserAbuseReport({ message, userId });
+
+    const expectedLoadAction = loadUserAbuseReport({
+      message: response.message,
+      reporter: response.reporter,
+      userId,
+    });
+
+    const calledAction = await sagaTester.waitFor(expectedLoadAction.type);
+    mockApi.verify();
+
+    expect(calledAction).toEqual(expectedLoadAction);
+  });
+
+  it('handles an empty reporter', async () => {
+    const userId = createUserAccountResponse({ id: 50 }).id;
+    const message = 'I have no reporter';
+    const response = createFakeUserAbuseReport({
+      message,
+      reporter: null,
+      userId,
+    });
+
+    mockApi
+      .expects('reportUser')
+      .once()
+      .returns(Promise.resolve(response));
+
+    _sendUserAbuseReport({ message, userId });
+
+    const expectedLoadAction = loadUserAbuseReport({
+      message: response.message,
+      reporter: response.reporter,
+      userId,
+    });
+
+    const calledAction = await sagaTester.waitFor(expectedLoadAction.type);
+    mockApi.verify();
+
+    expect(calledAction).toEqual(expectedLoadAction);
+  });
+
+  it('clears the error handler', async () => {
+    _sendUserAbuseReport();
+
+    const calledAction = await sagaTester.waitFor(CLEAR_ERROR);
+    expect(calledAction).toEqual(errorHandler.createClearingAction());
+  });
+
+  it('dispatches an error', async () => {
+    const error = new Error('some API error maybe');
+    mockApi
+      .expects('reportUser')
+      .returns(Promise.reject(error));
+
+    _sendUserAbuseReport();
+
+    const errorAction = errorHandler.createErrorAction(error);
+    const calledAction = await sagaTester.waitFor(errorAction.type);
+    expect(calledAction).toEqual(errorAction);
+  });
+
+  it('resets the state when an error occurs', async () => {
+    const userId = createUserAccountResponse({ id: 501 }).id;
+    const error = new Error('some API error maybe');
+    mockApi
+      .expects('reportUser')
+      .returns(Promise.reject(error));
+
+    _sendUserAbuseReport({ userId });
+
+    const abortAction = abortUserAbuseReport({ userId });
+    const calledAction = await sagaTester.waitFor(abortAction.type);
+    expect(calledAction).toEqual(abortAction);
+  });
+
+  it('throws an error if multiple reports are submitted for the same user', async () => {
+    const userId = createUserAccountResponse({ id: 50 }).id;
+
+    _sendUserAbuseReport({
+      message: 'This user is uncool!',
+      userId,
+    });
+
+    // Report the same add-on again; this will cause the reducer to throw
+    // an error and the saga should dispatch an error.
+    _sendUserAbuseReport({
+      message: 'Duplicate!',
+      userId,
+    });
+
+    await sagaTester.waitFor(SET_ERROR);
```

Comment:
What throws an error? I don't see any duplicate checking code in https://github.com/mozilla/addons-frontend/pull/4474/files#diff-725633e03c598c1ecedc6e82646a2eedR189 Did you check that this test fails when you remove the duplicate checking code?